### PR TITLE
Fede.step

### PR DESCRIPTION
**Pull request by federiva**
Implementing Step - Work in Progress

See [here](https://fomantic-ui.com/elements/step.html) for reference

**DoD**

- [x] Major project work has a corresponding task. If there’s no task for what you are doing, create it. Each task needs to be well defined and described.

- [x] Change has been tested (manually or with automated tests), everything runs correctly and works as expected. No existing functionality is broken.

- [ ] No new error or warning messages are introduced.

- [ ] All interaction with a semantic functions, examples and docs are written from the perspective of the person using or receiving it. They are understandable and helpful to this person.

- [ ] If the change affects code or repo sctructure, README, documentation and code comments should be updated. 

- [ ] All code has been peer-reviewed before merging into any main branch.

- [ ] All changes have been merged into the main branch we use for development (develop).

- [ ] Continuous integration checks (linter, unit tests) are configured and passed.

- [ ] Unit tests added for all new or changed logic.

- [ ] All task requirements satisfied. The reviewer is responsible to verify each aspect of the task.

- [ ] Any added or touched code follows our style-guide.

### CODE REVIEW

**Comment by federiva** [comment]
For the purpose of testing you can try the example in the steps function documentation

**Review by dokato — CHANGES_REQUESTED** [review]
Looks really good, great job 👍 See some minor comments below.

Still TODO:
- add example to `example/app.R` (and ask DB for deploy to our demo server)
- add simple unit tests

File: R/step.R [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+#' @param class (Optional) A character string with the semantic class to be
+#' added to the steps element.
+#' @seealso single_steps
+#' @examples 
+#' if (interactive()) {
+#'  library(shiny)
+#'  library(shiny.semantic)
+#'  ui <- semanticPage(
+#'  title = "Steps Example",
+#'  hr(),
```
[the remark this answers is not in the document]
Comment (federiva) [reply R/step.R:14]: done

File: R/step.R [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    )
+}
+
+
+#' @rdname toggle_step_state
+#' @export
+toggle_step_state <- function(id, state = TRUE, automatic_steps = TRUE, asis = FALSE) {
+    session <- shiny::getDefaultReactiveDomain()
+    # Make sure set_attribute_by_id works with namespaces (shiny modules)
+    id <- ifelse(
```
[the remark this answers is not in the document]
Comment (federiva) [reply R/step.R:134]: done

File: R/step.R [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+#' @export
+single_step <- function(id, title, description = NULL, icon_class = NULL,
+                        class = NULL) {
+    step_icon <- if (is.null(icon_class)) NULL else tags$i(class = paste(icon_class, "icon")) # nolint
+    step_description <- if (is.null(description)) NULL else shiny::div(
+        class = "description",
+        description
+    )
+    step_title <- shiny::div(class = "title", title)
+    step_class <- ifelse(
```
[the remark this answers is not in the document]
Comment (federiva) [reply R/step.R:111]: done

File: R/step.R [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+        inherits(session, "session_proxy") && !asis,
+        session$ns(id),
+        id
+    )
+    parameters <- list(
+        "step_id" = id,
+        "state" = state,
+        "automatic_steps" = automatic_steps)
+    session$sendCustomMessage("toggle_step_state", parameters)
+}
```
[the remark this answers is not in the document]
Comment (federiva) [reply R/step.R:144]: done

**Review by dokato — APPROVED** [review]
LGTM, great job @federiva !

**Pull request merged**